store/postgres/src/chain_head_listener.rs

```diff
@@ -173,9 +178,19 @@ impl ChainHeadUpdateListener {
 
                 // If there are subscriptions for this network, notify them.
                 if let Some(watcher) = watchers.read(&logger).get(&update.network_name) {
-                    debug!(logger, "sending chain head update"; "network" => &update.network_name);
-                    watcher.send();
-                    debug!(logger, "chain head update sent"; "network" => &update.network_name);
+                    // Due to a tokio bug, we must assume that the watcher can deadlock, see
+                    // https://github.com/tokio-rs/tokio/issues/4246.
+                    if !sending_to_watcher.load(atomic::Ordering::SeqCst) {
+                        let sending_to_watcher = sending_to_watcher.cheap_clone();
+                        let sender = watcher.sender.cheap_clone();
+                        tokio::task::spawn_blocking(move || {
```

Comment:
Can that lead to a large number of dead threads hanging around? Maybe we need another `watcher_is_live: Arc<AtomicBool>` that is set to `false` just before the call to send and set to `true` right afterwards where we do not try to send if `watcher_is_live` is false.

Reply:
I believe the `sending_to_watcher` is already serving as the check you propose.

Reply:
Yes, of course, you are right.